extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/KnativeEventsBindingRecorder.java

```diff
@@ -187,8 +226,51 @@ public void run() {
 
         Handler<RoutingContext> handler = new VertxRequestHandler(vertx.get(), rootPath, beanContainer, objectMapper,
                 eventsConfig,
-                defaultInvoker, typeTriggers, executor);
+                defaultInvoker, typeTriggers, invokersFilters, executor);
 
         return handler;
     }
+
+    private List<Predicate<CloudEvent>> filter(String functionName, CloudEventMapping mapping) {
+
+        if (mapping.attributes() == null || mapping.attributes().length == 0) {
+            return Collections.emptyList();
+        }
+        List<Predicate<CloudEvent>> filters = new ArrayList<>();
+        for (EventAttribute attribute : mapping.attributes()) {
+            Objects.requireNonNull(attribute.name(),
+                    "Attribute name of the EventAttribure on function " + functionName + " is required");
+            Objects.requireNonNull(attribute.value(),
+                    "Attribute name of the EventAttribure on function " + functionName + " is required");
+
+            if (!CE_ATTRIBUTES.contains(attribute.name())) {
+                filters.add(new CustomAttributeLiteralEqualsFilter(attribute.name(), attribute.value()));
+            } else {
+
+                filters.add(new CEAttributeLiteralEqualsFilter(attribute.name(), attribute.value()));
+            }
+        }
+
+        return filters;
+    }
+
+    private boolean hasSameFilters(List<Predicate<CloudEvent>> one, List<Predicate<CloudEvent>> two) {
```

Comment:
This seems to test whether intersection of those two list is non-empty. I don't think that's what we want. See my comment on `class WithOverlapingAttributeFilter`. We should test if one list is super-set of the another.

Reply:
Maybe what we want to check is that there exist two elements `ea1 ∊ one`, `ea2 ∊ two` such that `ea1.name = ea2.name` and `ea1.value ≠ ea2.value`.

If the above is true then the two lists are not conflicting.

Reply:
This is getting complicated. Maybe we could be satisfied with runtime conflict checking after all.

Reply:
maybe we could prevent at build time the duplicates in case someone defined the same filters so at least that would be covered, wdyt?

Reply:
Yeah, we should throw build time error only if there is guaranteed conflict. It happens when set of elements form one list is sub-set of the set of elements from the another. I described this as `Example B` in some other comment.

Reply:
```diff
diff --git a/extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/KnativeEventsBindingRecorder.java b/extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/KnativeEventsBindingRecorder.java
index 252fa53523..6b6e78759a 100644
--- a/extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/KnativeEventsBindingRecorder.java
+++ b/extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/KnativeEventsBindingRecorder.java
@@ -3,14 +3,7 @@ package io.quarkus.funqy.runtime.bindings.knative.events;
 import java.lang.reflect.Method;
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
+import java.util.*;
 import java.util.concurrent.Executor;
 import java.util.function.Predicate;
 import java.util.function.Supplier;
@@ -255,22 +248,15 @@ public class KnativeEventsBindingRecorder {
     }
 
     private boolean hasSameFilters(List<Predicate<CloudEvent>> one, List<Predicate<CloudEvent>> two) {
-        System.out.println(one + " " + two);
-        if (one == null && two == null) {
-            return true;
-        }
-
-        if (one != null && two != null) {
 
-            if (one.isEmpty() && two.isEmpty()) {
-                return true;
-            }
+        final List<Predicate<CloudEvent>> first = one != null ? one : Collections.emptyList();
+        final List<Predicate<CloudEvent>> second = two != null ? two : Collections.emptyList();
 
-            if (one.stream().anyMatch(item -> two.contains(item))) {
-                return true;
-            }
+        // empty set is sub-set of any set
+        if (first.size() <= 0 || second.size() <= 0) {
+            return true;
         }
 
-        return false;
+        return first.size() <= second.size() ? second.containsAll(first) : first.containsAll(second);
     }
 }
```

Reply:
thanks, will include these updates and push it by the end of today